src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/IRazorCodeActionHandler.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Microsoft.AspNetCore.Razor.LanguageServer.CodeActions.Models;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client.Capabilities;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.CodeActions
+{
+    [Parallel, Method("textDocument/codeAction", Direction.ClientToServer)]
+    internal interface IRazorCodeActionHandler :
```

Comment:
Things generally start to get spooky when we define our own handlers. Is this an area where O# hasn't responded to an addition to the spec yet or are we using `RazorCodeActionParams` to get `out-of-spec` info off this request?

Reply:
`out-of-spec`.

Specifically `RazorCodeActionParams` which has a `ExtendedCodeActionContext`, which has an out of spec `SelectionRange`.